resources/check-layout-th.js

```diff
@@ -249,4 +249,19 @@ window.checkLayout = function(selectorList, callDone = true)
         done();
 };
 
+// This function is a thin wrapper around `checkLayout` and simply reads the
+// `CHECK_LAYOUT_DELAY_MS` variable to add a delay. This global variable is not
+// intended to be set by browsers; instead, polyfills can set it to give
+// themselves time to apply changes before proceeding with assertions about the
+// layout. Tests that call this function at some point and do some other work
+// after should `await` it to avoid race conditions.
+window.checkLayoutDelayed = async function(selectorList, callDone = true) {
+    const delay = window.CHECK_LAYOUT_DELAY_MS;
+    if (typeof delay === 'number') {
+        console.log(`Waiting ${delay}ms before checking layout...`);
+        await new Promise(r => step_timeout(r, delay));
+    }
+    window.checkLayout(selectorList, callDone);
+}
```

Comment:
@jgerigmeyer @mfreed7 this new wrapper adds a delay (without calling `setTimeout` directly). To limit the scope of the changes, only the WPTs related to anchor positioning use it.

Reply:
Thanks. A few comments:
 - can you please move this wrapper to a function in `css/css-anchor-position/support/test-common.js`?
 - You could call it something like `checkLayoutForAnchorPos()`.
 - Would it suffice to, instead of `setTimeout`, do something like:
 ```javascript
await waitUntilNextAnimationFrame();
await waitUntilNextAnimationFrame();
```
That usually is enough to ensure all events have fired and layout has been completed after those events. And it avoids the "picking a delay" problem.